src/plugins/blocksmith/components/blocksmith/index.jsx

```diff
@@ -1385,9 +1388,29 @@ class Blocksmith extends React.Component {
         ...sidebarHolderStyle,
         transform: `scale(${isDocumentClick ? 0 : 1})`,
       },
-    }));
+    }), this.createPopoverToggleEvent);
+  }
+
+  createPopoverToggleEvent() {
+    const { isSidebarOpen } = this.state;
+
+    if (isSidebarOpen) {
+      document.addEventListener('click', this.handleTogglePopover);
+    } else {
+      document.removeEventListener('click', this.handleTogglePopover);
+    }
   }
 
+  handleTogglePopover(e) {
```

Comment:
@cohencodes lets call this handlePopoverClick, and add a docblock explaining why it is needed/what it does